ethers-core/src/types/transaction/response.rs

```diff
@@ -508,10 +498,44 @@ impl PartialOrd<Self> for TransactionReceipt {
     }
 }
 
+impl TransactionReceipt {
+    pub fn rlp(&self) -> Bytes {
```

Comment:
We still want to implement the trait right? Functionality seems fine. Would the test case below fail w/o this PR?

Reply:
I think I didn't implement the trait because Transaction seemed to not implement the trait (I'll check again). Regardless, it's pretty easy to implement the trait. The new test case would fail w/o the PR because the old rlp encoding did not include the transaction type.

Reply:
Got it! Helpful context. Please add the trait implementation, and if you see other structs that are missing it, please also go for these as well. Thanks for taking this on!!!

Reply:
The traits should be implemented now for both Transaction and TransactionReceipt. I didn't implement Decodable for TransactionReceipt because I feel like I don't have enough familiarity with the documentation of all the structs and functions involved.